src/bootstrap/test.rs

```diff
@@ -1524,9 +1524,7 @@ impl Step for RustcGuide {
     fn run(self, builder: &Builder<'_>) {
         let src = builder.src.join("src/doc/rustc-guide");
         let mut rustbook_cmd = builder.tool_cmd(Tool::Rustbook);
-        builder.run(rustbook_cmd
-                       .arg("linkcheck")
-                       .arg(&src));
+        try_run(builder, rustbook_cmd.arg("linkcheck").arg(&src));
```

Comment:
We'll definitely want `try_run_quiet`  here as `try_run` eats the command's output (a horrible and non-obvious default, I know -- we should fix that).

Reply:
Wait so `quiet` makes it *not* eat the output? That's, like, the opposite of what the name indicates.^^

Reply:
Yep, `try_run` internally calls `run_silent` which will eat all the output unconditionally, I believe.

Reply:
The one in `lib.rs` does, yeah. The one in `test.rs` calls `builder.run`.

So `Builder::try_run` should maybe just be renamed to `try_run_silent`, like the free-standing function it calls?

Reply:
Fun times, there's both free-standing `run_silent` (called by `Builder::run`) and `run_suppressed` (called by `Builder::run_quiet`).

Reply:
Actually, @Mark-Simulacrum are you sure about this? [`try_run_silent`](https://github.com/rust-lang/rust/blob/f23a5f208de7fba983bfaaa8d60f1208d5780a98/src/build_helper/lib.rs#L59) calls [`Command::status`](https://doc.rust-lang.org/nightly/std/process/struct.Command.html#method.status) which says that

> By default, stdin, stdout and stderr are inherited from the parent.

I have no idea why it is called "silent". There is no comment explaining that.

In contrast, `run_suppressed` (the helper behind `Builder::run_quiet`) calls `Command::output`, which does capture the output, and then throws it away (unless there was an error).

Reply:
See https://github.com/rust-lang/rust/pull/63196: I think you are wrong and `try_run` does *not* eat the commends output. In contrast, `try_run_quiet` *does* eat (suppress) the output.

Reply:
Hm, you could definitely be right! I don't have a good handle on these functions beyond "too many and bad naming" :)